lib/Schema.ts

```diff
@@ -271,11 +271,11 @@ interface AttributeDefinition {
 	schema?: AttributeType | AttributeType[] | AttributeDefinition | AttributeDefinition[] | SchemaDefinition | SchemaDefinition[];
 	default?: ValueType | (() => ValueType);
 	forceDefault?: boolean;
-	validate?: ValueType | RegExp | ((value: ValueType) => boolean);
+	validate?: ValueType | RegExp | ((value: ValueType) => boolean|Promise<boolean>);
```

Comment:
Let's keep spacing consistent. This **should** be a lint rule. If you'd like to look into how to enable it as a lint rule that'd be awesome as well (but no pressure).

```suggestion
	validate?: ValueType | RegExp | ((value: ValueType) => boolean | Promise<boolean>);
```